### PR TITLE
159 refactor for receiver agent for horizontal scaling

### PR DESCRIPTION
**Pull request by mbrzus**
# Overview

This pull request introduces a Kubernetes-enabled version of our system, which has been adapted to run on Minikube locally. This adaptation aligns with modern deployment practices, aiming to enhance scalability and manageability.

# Implementation

The system architecture mirrors our existing Docker setup with key components such as the frontend, backend, PostgreSQL database, internal Orthanc server, receiver, study handler, and the example product. Key changes and enhancements include:

- Modular Design: The receiver and study handler have been restructured for better decoupling and to support horizontal scaling effectively.
- Automated Kubernetes Job Management: Introduces dynamic creation of Kubernetes Jobs for processing studies and products, facilitating better resource management and scalability.
- Configuration Management: Utilizes Kubernetes primitives like Deployments, Services, Ingresses, and Jobs extensively, alongside appropriate RBAC (Role-Based Access Control) settings to ensure secure and efficient operations.

# Usage

The new Kubernetes functionality can be deployed and managed using Minikube. Detailed instructions and step-by-step guides are provided in the README_KUBES.md file, which includes:

Setup instructions for Minikube and necessary Kubernetes configurations.
Deployment commands for all Kubernetes resources.
Guidelines on how to scale components and handle dynamic job creation.
# Testing

Extensive manual testing was conducted to ensure each component functions correctly within the Kubernetes environment. This includes testing data flow between components, job creation and execution, and the responsiveness of the system under different loads.

# Problems Faced

Transitioning to a Kubernetes-based deployment presented several challenges:

- Networking Issues: We encountered difficulties in managing network communications between pods and configuring external access via HTTP. These were addressed by refining our service and ingress configurations.
- Learning Curve: Our team's initial unfamiliarity with Kubernetes required us to engage in comprehensive learning and experimentation, which slowed early progress but significantly built our internal expertise.

### CODE REVIEW

File: Kubernetes/Dockerfile_study_handler
Comment (powersaudrey25) [inline Kubernetes/Dockerfile_study_handler:1]: this file and Dockerfile_Study are duplicates we only need to keep one

File: Dockerfile_OrthancReciever_old
Comment (powersaudrey25) [inline Dockerfile_OrthancReciever_old:1]: I think we can delete this

File: Kubernetes/Dockerfile_study_handler
Comment (mbrzus) [reply Kubernetes/Dockerfile_study_handler:1]: deleted the Dockerfile_study_handler

File: Dockerfile_OrthancReciever_old
Comment (mbrzus) [reply Dockerfile_OrthancReciever_old:1]: I agree. deleted

**Review by powersaudrey25 — APPROVED** [review]

**Pull request merged**